### ISSUE TITLE
Feature: Support Blazor Server

### ISSUE THREAD

**Issue by mavaddat** [KristofferStrube/Blazor.Popper#4]
### Description

`CreatePopperAsync` results in an exception, 

> An exception occurred executing JS interop: Deserialization of interface types is not supported. Type 'Microsoft.JSInterop.IJSInProcessObjectReference'. Path: $ | LineNumber: 0 | BytePositionInLine: 1.. See InnerException for more details.

with inner exception,

> Deserialization of interface types is not supported. Type 'Microsoft.JSInterop.IJSInProcessObjectReference'.

Specifically, this exception is thrown on https://github.com/KristofferStrube/Blazor.Popper/blob/6fcc66ec82c10082883c2522f418049d28654481/src/KristofferStrube.Blazor.Popper/Popper.cs#L25C45-L25C85:

```csharp
/* Line 25 below, KristofferStrube.Blazor.Popper/Popper.cs */
var popperWrapper = await jSRuntime.InvokeAsync<IJSInProcessObjectReference>("import", "./_content/KristofferStrube.Blazor.Popper/KristofferStrube.Blazor.popper.js");
```

It should also be a problem on [the other overloaded version of the same method](https://github.com/KristofferStrube/Blazor.Popper/blob/6fcc66ec82c10082883c2522f418049d28654481/src/KristofferStrube.Blazor.Popper/Popper.cs#L18), though I have not tested it yet.

### Reproduction Steps

1. Re-create https://github.com/KristofferStrube/Blazor.Popper/blob/main/samples/KristofferStrube.Blazor.Popper.Samples/Pages/VirtualElementReference.razor using .NET 8 (8.0.100-rc.2.23502.2), Microsoft.AspNetCore.App 8.0.0-rc.2.23480.2 within a MAUI Blazor hybrid application
2. Run the solution.
3. Step through to the line https://github.com/KristofferStrube/Blazor.Popper/blob/6fcc66ec82c10082883c2522f418049d28654481/samples/KristofferStrube.Blazor.Popper.Samples/Pages/VirtualElementReference.razor#L38C13-L38C89  
    ```csharp
    instance = await Popper.CreatePopperAsync(virtualElement, tooltip, options);
    ```

### Expected behavior

Injection of the Popper.js JavaScript interop (`./_content/KristofferStrube.Blazor.Popper/KristofferStrube.Blazor.popper.js`) into the Blazor page.

### Actual behavior

`Popper.CreatePopperAsync` fails with `Deserialization of interface types is not supported` exception.

### Regression?

The [current sample implementation](https://kristofferstrube.github.io/Blazor.Popper/VirtualElementReference) is working correctly, which suggests that a breaking change in ASP.NET stopped the de-serialization of the `Microsoft.JSInterop.IJSInProcessObjectReference` interface.

### Known Workarounds

None known.

### Configuration

Microsoft Windows 10.0.19045, OSDImageName is Windows 10 x64 21H2 v8.0.
The version of .NET I am using is .NET 8.0 (8.0.100-rc.2.23502.2). Microsoft.AspNetCore.Components 8.0.0-rc.2.23480.2
The target frameworks I am using are net8.0-android, net8.0-ios, net8.0-maccatalyst, and net8.0-windows10.0.19041.0.
The supported OS platform versions are 14.2 for iOS, 14.0 for MacCatalyst, 24.0 for Android, 10.0.17763.0 for Windows, and 6.5 for Tizen.
The application title is Bombardier XML Splitter and the application ID is com.bombardier.mauixmlsplitter.
The project URL is <https://github.com/Technical-Publications-Bombardier/BaXmlSplitter> and the repository type is git.

### Other information

See related issues:

https://github.com/dotnet/aspnetcore/issues/29981
https://github.com/dotnet/aspnetcore/issues/45440

**Comment by mavaddat** [comment]
I have not been able to reproduce this issue with [upgrading the source package project and sample project to .NET 8.0](https://github.com/mavaddat/Blazor.Popper). This suggests that there is something with .NET MAUI that interferes with the ASP .NET JS interop. Investigating this now.

**Comment by KristofferStrube** [comment]
This library only supports Blazor WASM as stated in the README. I can add support for Blazor Server and through that Blazor Hybrid in MAUI if you would like.

**Comment by mavaddat** [comment]
> This library only supports Blazor WASM as stated in the README.

I am just getting started with Blazor, so I thought server and WASM were just two use-cases &mdash; I did not realize there was a difference in the underlying technology.

> I can add support for Blazor Server and through that Blazor Hybrid in MAUI if you would like.

That would be really great. I like what you have done and I would appreciate the help.

**Comment by KristofferStrube** [comment]
Hey @mavaddat,

I still need to set up a contribution guideline for this repository, so I understand the confusion. There are better ways to contribute to open-source repositories. I will get a contribution guideline in place soon.

Until then, I recommend that you read the following two articles [Open Source Contribution Etiquette](https://tirania.org/blog/archive/2010/Dec-31.html) and [Don't "Push" Your Pull Requests](https://www.igvita.com/2011/12/19/dont-push-your-pull-requests/).

Again, no harm done. It's just not the best way to contribute. I'm happy that you are interested in contributing.

**Comment by mavaddat** [comment]
OK, no problem. I will await the guideline.

**Comment by KristofferStrube** [comment]
Hey @mavaddat sorry for the long wait. I have guidelines in place now: https://github.com/KristofferStrube/Blazor.Popper/blob/main/CONTRIBUTING.md

And I will very shortly give a review of your original PR so that we can get it down to something that can be merged.

**Comment by xfxJeroen** [comment]
> Hey @mavaddat sorry for the long wait. I have guidelines in place now: https://github.com/KristofferStrube/Blazor.Popper/blob/main/CONTRIBUTING.md
> 
> And I will very shortly give a review of your original PR so that we can get it down to something that can be merged.

Does this mean there will be Blazor Server support?

**Comment by KristofferStrube** [comment]
You can see that the PR is still open here: https://github.com/KristofferStrube/Blazor.Popper/pull/5 meaning that it hasn't been merged into master yet nor been released.

I will try to prioritize support for Blazor Server very soon, but I wanted to give someone the option to contribute first.

**Comment by KristofferStrube** [comment]
Fixed in #6

@xfxJeroen you might be interested in checking out version 0.6.0 on NuGet for Blazor Server support. :)
